libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -182,12 +185,16 @@ internal class MapboxTripSession(
         if (state == TripSessionState.STOPPED) {
             return
         }
-        tripService.stopService()
-        stopLocationUpdates()
-        ioJobController.job.cancelChildren()
-        mainJobController.job.cancelChildren()
-        reset()
-        state = TripSessionState.STOPPED
+
+        stopSessionJob = ThreadController.getMainScopeAndRootJob().scope.launch {
```

Comment:
```suggestion
        stopSessionJob = threadController.getMainScopeAndRootJob().scope.launch {
```

nit: kind of odd to pass in an instance of the singleton and then not use it.

Reply:
fixed